src/main/java/com/snowflake/kafka/connector/internal/SnowflakeSinkServiceV1.java

```diff
@@ -617,6 +630,10 @@ private void startCleaner(long recordOffset, SnowflakeTelemetryPipeCreation pipe
                     e.getMessage(),
                     e.getStackTrace());
                 telemetryService.reportKafkaFatalError(e.getMessage());
+                if (maxCleanerRetries >= 0 &&  pipeStatus.cleanerRestartCount.get() == maxCleanerRetries) {
+                  failedCleanerException.set(e);
+                  break;
+                }
```

Comment:
Is this true?

Reply:
Can you fix the tests which are failing? https://github.com/snowflakedb/snowflake-kafka-connector/pull/453

Reply:
> Can you fix the tests which are failing? #453

@sfc-gh-japatel , I looked at the failing tests and noticed this one =>

```
19:26:11  === Sleep 60 secs before verify result in Snowflake DB ===
  File "test_verify.py", line 550, in execution
    driver.verifyWithRetry(test.verify, r)
  File "test_verify.py", line 112, in verifyWithRetry
    func(round)
  File "/home/runner/work/snowflake-kafka-connector/snowflake-kafka-connector/test/test_suit/test_pressure_restart.py", line 50, in verify
    self.driver.createConnector(self.getConfigFileName(), self.nameSalt)
  File "test_verify.py", line 337, in createConnector
    print(datetime.now().strftime("%H:%M:%S "), json.loads(r.content.decode("utf-8"))["name"], r.status_code)
19:27:11  === generate sink connector rest reqeuest from ./rest_request_template ===
```

Seems unrelated to the changes in this PR.. Or am I looking at the wrong test failures?

Reply:
> Is this true?

hmm I think it should be `if (maxCleanerRetries >= 0 &&  pipeStatus.cleanerRestartCount.get() >= maxCleanerRetries) {`

Because, resetCleanerFiles() always increments the cleanerRestartCount and if `maxCleanerRetries` is set to 0, then we will never reach the if condition. WDYT?

Reply:
@sfc-gh-japatel did you get a chance to look at my comments? Thanks!

Reply:
I was talking about the Integration test. 
`testCleanerRecover` is the failing test.

Reply:
I tried fixing the test but didnt spend too much time on it, can you run the test and fix the PR? I can help you with the profile.json file which you might not have.

Reply:
Sure @sfc-gh-japatel  I would fix it. Can you plz provide with the profile.json ?